Undefined symbols break ompi v5.0.x build configured with `--with-sge`

## Related PR
https://github.com/openpmix/prrte/pull/1538

## Background information
I'm trying to build ompi v5.0.x branch. If I `./configure ... --with-sge ... && make`, I get the error:
```
...
make[4]: Entering directory `/home/ec2-user/ompi/3rd-party/prrte/src/mca/ras/gridengine'
  CC       ras_gridengine_component.lo
  CC       ras_gridengine_module.lo
ras_gridengine_component.c:60:9: warning: implicit declaration of function 'PRTE_MCA_BASE_MAKE_VERSION'; did you mean 'PMIX_MCA_BASE_MAKE_VERSION'? [-Wimplicit-function-declaration]
         PRTE_MCA_BASE_MAKE_VERSION(component,
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
         PMIX_MCA_BASE_MAKE_VERSION
...
ras_gridengine_component.c:76:12: warning: implicit declaration of function 'prte_mca_base_component_var_register'; did you mean 'pmix_mca_base_component_var_register'? [-Wimplicit-function-declaration]
     (void) prte_mca_base_component_var_register(c, "priority",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            pmix_mca_base_component_var_register
...
```

With a quick code search([search 1](https://github.com/search?q=org%3Aopenpmix+PRTE_MCA_BASE_MAKE_VERSION&type=code), [search 2](https://github.com/search?q=org%3Aopenpmix+prte_mca_base_component_var_register&type=code)), I find both symbols are indeed undefined and should be cleaned up.


### ompi version
https://github.com/open-mpi/ompi/commit/20745d8c0922e63df694f826aa7f4729cd82217c

### What version of the PMIx Reference RTE (PRRTE) are you using? (e.g., v2.0, v3.0, git master @ hash, etc.)
https://github.com/openpmix/prrte/commit/a3e81f2efbc6abc420a620d219dd7d332a8e9468

##### What version of PMIx are you using? (e.g., v4.2.0, git branch name and hash, etc.)
https://github.com/openpmix/openpmix/commit/b4a55542880b889306b8b99ebce1fcd9609db0b7


### Please describe the system on which you are running

* Operating system/version: Ubuntu20.04
* Computer hardware: AWS EC2 c5n.18xlarge
* Network type: EFA fabric
* Verified ompi prerequisites: https://docs.open-mpi.org/en/v5.0.x/developers/prerequisites.html

-----------------------------

## Details of the problem

Steps to reproduce
```shell
git clone --recursive https://github.com/open-mpi/ompi.git
cd ompi
git checkout v5.0.x
git submodule update --init --recursive
./autogen.pl
./configure --prefix=~/ompi/install --with-sge --without-verbs --with-libfabric=/opt/amazon/efa --disable-man-pages --with-libevent=external --with-hwloc=external --disable-builtin-atomics
make -j32 # error
```

Note that removing `--with-sge` bypasses the issue, i.e.
```shell
...
./configure --prefix=~/ompi/install --without-verbs --with-libfabric=/opt/amazon/efa --disable-man-pages --with-libevent=external --with-hwloc=external --disable-builtin-atomics
make -j32 # this works
```

--- a/src/mca/ras/gridengine/ras_gridengine_component.c
+++ b/src/mca/ras/gridengine/ras_gridengine_component.c
@@ -57,7 +57,7 @@ prte_mca_ras_gridengine_component_t prte_mca_ras_gridengine_component = {
     .super = {
         PRTE_RAS_BASE_VERSION_2_0_0,
         .pmix_mca_component_name = "gridengine",
-        PRTE_MCA_BASE_MAKE_VERSION(component,
+        PMIX_MCA_BASE_MAKE_VERSION(component,
                                    PRTE_MAJOR_VERSION,
                                    PRTE_MINOR_VERSION,
                                    PMIX_RELEASE_VERSION),
@@ -73,19 +73,21 @@ static int prte_ras_gridengine_register(void)
     pmix_mca_base_component_t *c = &prte_mca_ras_gridengine_component.super;
 
     prte_mca_ras_gridengine_component.priority = 100;
-    (void) prte_mca_base_component_var_register(c, "priority",
+    (void) pmix_mca_base_component_var_register(c, "priority",
                                                 "Priority of the gridengine ras component",
+                                                PMIX_MCA_BASE_VAR_TYPE_INT,
                                                 &prte_mca_ras_gridengine_component.priority);
 
     prte_ras_gridengine_verbose = 0;
-    (void) prte_mca_base_component_var_register(c, "verbose",
-                                                "Enable verbose output for the gridengine ras component",
-                                                &prte_ras_gridengine_verbose);
+    (void) pmix_mca_base_component_var_register(
+        c, "verbose", "Enable verbose output for the gridengine ras component",
+        PMIX_MCA_BASE_VAR_TYPE_INT, &prte_ras_gridengine_verbose);
 
     prte_mca_ras_gridengine_component.show_jobid = false;
-    (void) prte_mca_base_component_var_register(c, "show_jobid",
+    (void) pmix_mca_base_component_var_register(c, "show_jobid",
                                                 "Show the JOB_ID of the Grid Engine job",
-                                                 &prte_mca_ras_gridengine_component.show_jobid);
+                                                PMIX_MCA_BASE_VAR_TYPE_BOOL,
+                                                &prte_mca_ras_gridengine_component.show_jobid);
 
     return PRTE_SUCCESS;
 }

--- a/src/mca/schizo/hydra/schizo_hydra_component.c
+++ b/src/mca/schizo/hydra/schizo_hydra_component.c
@@ -34,7 +34,7 @@ prte_schizo_hydra_component_t prte_schizo_hydra_component = {
         .base_version = {
             PRTE_MCA_SCHIZO_BASE_VERSION_1_0_0,
             \.pmix_mca_component_name = "hydra",
-            PRTE_MCA_BASE_MAKE_VERSION(component, PRTE_MAJOR_VERSION, PRTE_MINOR_VERSION,
+            PMIX_MCA_BASE_MAKE_VERSION(component, PRTE_MAJOR_VERSION, PRTE_MINOR_VERSION,
                                         PMIX_RELEASE_VERSION),
             .mca_query_component = component_query,
         },

